src/diffusers/models/transformers/transformer_2d.py

```diff
@@ -305,14 +307,16 @@ def _init_patched_inputs(self, norm_type):
 
         # PixArt-Alpha blocks.
         self.adaln_single = None
-        self.use_additional_conditions = False
         if self.config.norm_type == "ada_norm_single":
-            self.use_additional_conditions = self.config.sample_size == 128
+            self.use_additional_conditions = self.config.sample_size == 128 and self.use_additional_conditions
             # TODO(Sayak, PVP) clean this, for now we use sample size to determine whether to use
             # additional conditions until we find better name
             self.adaln_single = AdaLayerNormSingle(
                 self.inner_dim, use_additional_conditions=self.use_additional_conditions
             )
+        else:
```

Comment:
I don't understand the logic here so I think something might be wrong, basically, we throw a warning as long as the norm_type is not ada_norm_single 

```python
if self.config.norm_type == "ada_norm_single":
     ...
else:
    warn about use_additional_condition?
```

Reply:
Oh yes, I think it could be:

```python
elif self.config.norm_type != "ada_norm_single" and self.config. use_additional_conditions
```

?